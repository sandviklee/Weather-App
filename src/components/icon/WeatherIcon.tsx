export type WeatherStatus =
  | "clearsky_day"
  | "clearsky_night"
  | "cloudy"
  | "fair_day"
  | "fog"
  | "heavyrain"
  | "heavyrainandthunder"
  | "heavyrainshowers_day"
  | "lightrain"
  | "lightrainshowers_day"
  | "lightrain"
  | "lightrainshowers_day"
  | "no_weather"
  | "partlycloudy_day"
  | "partlycloudy_night"
  | "rain"
  | "rainshowers_day"
  | "rainshowers_night"
  | "rainshowersandthunder_day"
  | "sleet"
  | "snow"
  | "snowandthunder";

export const norwegianWeatherStatusMapping = {
  clearsky_day: "Klarvær",
  clearsky_night: "Klarvær",
  cloudy: "Overskyet",
  fair_day: "Delvis skyet",
  fog: "Tåke",
  heavyrain: "Kraftig regn",
  heavyrainandthunder: "Kraftig regn og torden",
  heavyrainshowers_day: "Kraftige regnbyger",
  lightrain: "Lett regn",
  lightrainshowers_day: "Lette regnbyger",
  no_weather: "Ingen værdata",
  partlycloudy_day: "Delvis skyet",
  partlycloudy_night: "Delvis skyet",
  rain: "Regn",
  rainshowers_day: "Regnbyger",
  rainshowers_night: "Regnbyger",
  rainshowersandthunder_day: "Regnbyger og torden",
  sleet: "Sludd",
  snow: "Snø",
  snowandthunder: "Snø og torden",
};

interface WeatherIconProps {
  status: WeatherStatus;
  size?: number;
}

const WeatherIcon = ({ status, size = 24 }: WeatherIconProps) => {
  const iconSrc = `/project1/yr-icons/${status}.svg`;

  return (
    <img
      src={iconSrc}
      alt="WeatherStatus"
      width={size}
      height={size}
      style={{ margin: "0px" }}
      draggable={false}
    />
  );
};

export default WeatherIcon;
