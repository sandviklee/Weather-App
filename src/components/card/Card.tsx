import styles from "./Card.module.css";
import WeatherIcon from "../icon/WeatherIcon";

interface CardProps {
  location: string;
  selected?: boolean;
  temperature: number;
  nightTemperature: number;
}

const Card = ({
  location,
  selected = false,
  temperature,
  nightTemperature,
}: CardProps) => {
  return (
    <div className={styles.card}>
      <div
        className={styles.container}
        style={
          selected
            ? {
                border: "3px solid #000000",
              }
            : {
                border: "3px solid transparent",
              }
        }
      >
        <h1 className={styles.location}>{location}</h1>
        <WeatherIcon status="heavyrain" size={60} />
        <div className={styles.temperatures}>
          <p className={styles.temperature}>{temperature}°</p>
          <p className={styles.night_temperature}>{nightTemperature}°</p>
        </div>
      </div>
      {selected && <div className={styles.selected}></div>}
    </div>
  );
};

export default Card;
