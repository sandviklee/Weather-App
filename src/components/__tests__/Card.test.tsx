import { render, screen } from '@testing-library/react';
import Card from '../card/Card.tsx';
import { WeatherStatus } from '../icon/WeatherIcon.tsx';

const weatherStatus: WeatherStatus = 'clearsky_day';

const mockData = {
    location: 'Test Location',
    selected: false,
    temperature: 20,
    next12hours: 15,
    status: weatherStatus,
};

describe('Card component', () => {
    test('renders correctly', () => {
        render(<Card {...mockData} />);
        
        const location = screen.getByText(mockData.location);
        expect(location).toBeInTheDocument();
        const temperature = screen.getByText(`${mockData.temperature}°`);
        expect(temperature).toBeInTheDocument();
        const next12hours = screen.getByText(`${mockData.next12hours}°`);
        expect(next12hours).toBeInTheDocument();
    });

    test('applies the correct styles when selected', () => {
        render(<Card {...{ ...mockData, selected: true }} />);
        const cardContainer = screen.getByText(mockData.location).parentElement;
        expect(cardContainer).toHaveStyle({
            backgroundColor: "rgb(0, 0, 0)",
            color: "rgb(255, 255, 255)",
        });
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<Card {...mockData} />);
        expect(asFragment()).toMatchSnapshot();
    });
});