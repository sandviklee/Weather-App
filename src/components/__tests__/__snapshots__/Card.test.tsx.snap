// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_card_3c4546"
  >
    <div
      class="_container_3c4546"
      style="background-color: white;"
    >
      <h1
        class="_location_3c4546"
      >
        Test Location
      </h1>
      <img
        alt="WeatherStatus"
        draggable="false"
        height="24"
        src="/project1/yr-icons/clearsky_day.svg"
        style="margin: 0px;"
        width="24"
      />
      <div
        class="_temperatures_3c4546"
      >
        <p
          class="_temperature_3c4546"
        >
          20°
        </p>
        <p
          class="_night_temperature_3c4546"
        >
          15°
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
